File: InitialTrainingLibrary/InitialTrainingLibrary/dysq/DateValidator/MyDate.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using InitialTrainingLibrary.dysq.DateValidator.Days;
using InitialTrainingLibrary.Interfaces;

namespace InitialTrainingLibrary.dysq.DateValidator
{
    public class MyDate : IDateValidator
    {

        public bool DateValidate(int year, int month, int day)
        {
        
            if (IsMonthValid(month))
            {
                if (IsDayValidOf31(day) && Months.GetMonths31Days().Any(_month => _month == month))
                    return true;

                if (IsDayValidOf30(day) && Months.GetMonths30Days().Any(_month => _month == month))
                    return true;
                
                if (IsDayValidOf28(day))
                    return Months.Is28DaysMonth(month);   
            }

            return false;
        }

        public string GetName()
        {
            return "Dysq";
        }


        private bool IsMonthValid(int month)
        {
            return Months.IsMonthCorrect(month);
        }

        private bool IsDayValidOf31(int day)
        {
            return Days31ofMonths.IsCorrect(day);
        }

        private bool IsDayValidOf30(int day)
        {
            return Days30ofMonths.IsCorrect(day);
        }

        private bool IsDayValidOf28(int day)
        {
            return Days28ofMonth.IsCorrect(day);
        }
    }
}

File: Eagles/Eagles Reloaded/TrainingOneHomework/Mikolaj/classes/MyDate.php
<?php
class MyDate
{
	protected $months = array(
			0,
			1  => 31,
			2  => 28,
			3  => 31,
			4  => 30,
			5  => 31,
			6  => 30,
			7  => 31,
			8  => 31,
			9  => 30,
			10 => 31,
			11 => 30,
			12 => 31
		);

	
	public function check($date = array())
	{
		$index = (intval($date['month']) % 20 <= 12)? intval($date['month']) % 20 : 0;
		//leap year
		if(intval($date['year']) % 4 == 0)    $this->months[2] += 1;
		//ordinary date
		if($this->months[$index]>=$date['day'] && $date['day'] > 0 )
			return true;
		else 
			return false;

	}
}

?>